Move activityStateChangeID from transaction scope to main frame scope in CommitLayerTree

ruthvikkonda opened the pull request:
#### 87d9a6b1f9f21b312f69c79b7b426a045263f418
<pre>
Move activityStateChangeID from transaction scope to main frame scope in CommitLayerTree
<a href="https://bugs.webkit.org/show_bug.cgi?id=301114">https://bugs.webkit.org/show_bug.cgi?id=301114</a>
<a href="https://rdar.apple.com/163054763">rdar://163054763</a>

Reviewed by Matt Woodrow.

Right now RemoteLayerTreeDrawingAreaProxy::commitLayerTree currently receives a Vector&lt;std::pair&lt;RemoteLayerTreeTransaction,
RemoteScrollingCoordinatorTransaction&gt;&gt;, one pair per root frame. However, some aspects of the transaction are
scoped to the root frame. Some are scoped to the page. And some are scoped to the main frame. We want to accordingly
introduce some separation/organization here to reduce duplication and unnecessary isMainFrameProcessTransaction() checks.

This PR is the first of many that starts this organization/migration. In this PR, we just move one member activityStateChangeID from
transaction scope to main frame scope. It gets accordingly removed from the isMainFrameProcessTransaction() check
on UIProcess side.

No new functionality is added. This is a simple refactoring. Existing test coverage is enough.
As a result, no new tests are added.

* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeCommitBundle.h: Added.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
(WebKit::RemoteLayerTreeTransaction::activityStateChangeID const): Deleted.
(WebKit::RemoteLayerTreeTransaction::setActivityStateChangeID): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.messages.in:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):

Canonical link: <a href="https://commits.webkit.org/301885@main">https://commits.webkit.org/301885@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5dc9462e61ef9d0dcc6becc057d31bd4fd54744b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127266 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46914 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38045 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134330 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78821 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2138e8a8-62f9-4df7-b68d-39dfe52e3d0c) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47524 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55435 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96854 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64885 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e2aaec7b-144b-49e7-aafc-0eed576f831e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130214 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/38031 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113979 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77351 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f870a2b1-f9e2-4c0a-9cf7-7e02d043f0a4) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36912 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77710 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107890 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32476 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136813 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53925 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41546 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105372 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54436 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110332 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105057 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26803 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50577 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/29044 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51488 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53862 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59949 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/53094 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56554 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54855 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mattwoodrow commented on Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeCommitBundle.h:
```diff
@@ -0,0 +1,45 @@
+namespace WebKit {
+
+struct RemoteLayerTreeCommitBundle {
+    Vector<std::pair<RemoteLayerTreeTransaction, RemoteScrollingCoordinatorTransaction>> transactions;
```
Could typedef this pair as being the 'RootFrameData'?

ruthvikkonda replied:
Good idea!

ruthvikkonda committed:
Move activityStateChangeID from transaction scope to main frame scope in CommitLayerTree
https://bugs.webkit.org/show_bug.cgi?id=301114
rdar://163054763

Reviewed by Matt Woodrow.

Right now RemoteLayerTreeDrawingAreaProxy::commitLayerTree currently receives a Vector<std::pair<RemoteLayerTreeTransaction,
RemoteScrollingCoordinatorTransaction>>, one pair per root frame. However, some aspects of the transaction are
scoped to the root frame. Some are scoped to the page. And some are scoped to the main frame. We want to accordingly
introduce some separation/organization here to reduce duplication and unnecessary isMainFrameProcessTransaction() checks.

This PR is the first of many that starts this organization/migration. In this PR, we just move one member activityStateChangeID from
transaction scope to main frame scope. It gets accordingly removed from the isMainFrameProcessTransaction() check
on UIProcess side.

No new functionality is added. This is a simple refactoring. Existing test coverage is enough.
As a result, no new tests are added.

* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeCommitBundle.h: Added.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
(WebKit::RemoteLayerTreeTransaction::activityStateChangeID const): Deleted.
(WebKit::RemoteLayerTreeTransaction::setActivityStateChangeID): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.messages.in:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTreeTransaction):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updateRendering):

Canonical link: https://commits.webkit.org/301885@main
